File: src/nougen_shards/brain_scan/reporters.py
import json
from typing import List
from .candidate import CandidateFile
from .importer import ImportResult

def print_scan_report(candidates: List[CandidateFile], as_json: bool = False):
    """Formats and prints the result of a Brain Scan."""
    if as_json:
        data = [{"path": str(c.path), "tool": c.tool, "score": c.score_tier, "size_mb": c.size_mb} for c in candidates]
        print(json.dumps(data, indent=2))
        return

    print("\n🧠 NouGenShards Brain Scan\n")
    
    high = [c for c in candidates if c.score_tier == "high"]
    med = [c for c in candidates if c.score_tier == "medium"]
    
    tools = {}
    for c in candidates:
        tools[c.tool] = tools.get(c.tool, 0) + 1

    print("High-confidence AI memory:")
    for tool, count in tools.items():
        if tool != "unknown":
            print(f"  .{tool:<12} found   {count} files likely")
            
    print("\nProject context:")
    for c in [c for c in candidates if c.is_project_context][:5]:
        print(f"  {c.path.name}")
    if len([c for c in candidates if c.is_project_context]) > 5:
        print("  ... and more.")

    print("\nSkipped danger zones:")
    print("  .ssh         skipped by default")
    print("  .aws         skipped by default")
    print("  .azure       skipped by default")
    print("  .config      skipped by default")

    print(f"\nEstimated new shards: {len(high) * 2 + len(med)}") # Heuristic
    print("Cloud calls: 0")
    print("Files modified: 0")
    print("\nNothing imported.")
    print("Run: nougen brain import --confirm")

def print_import_report(result: ImportResult, dry_run: bool, as_json: bool = False):
    """Formats and prints the result of a Brain Import."""
    if as_json:
        print(json.dumps({
            "dry_run": dry_run,
            "files_scanned": result.files_scanned,
            "records_parsed": result.records_parsed,
            "shards_created": result.shards_created,
            "duplicates_skipped": result.duplicates_skipped,
            "secrets_redacted": result.secrets_redacted
        }, indent=2))
        return

    if dry_run:
        print("\n🧠 NouGenShards Brain Import (Dry Run)\n")
        print(f"Files to scan: {result.files_scanned}")
        print(f"Estimated records: {result.records_parsed}")
        print("\nRun: nougen brain import --confirm to write to memory.")
    else:
        print("\n🧠 NouGenShards Brain Import Complete\n")
        print(f"Files scanned:      {result.files_scanned}")
        print(f"Records parsed:     {result.records_parsed}")
        print(f"Shards created:     {result.shards_created}")
        print(f"Duplicates skipped: {result.duplicates_skipped}")
        print(f"Secrets redacted:   {result.secrets_redacted}")
        print("\n✅ Local memory enriched.")

File: ts/src/nougen_shards/brain_scan/reporters.ts
/**
 * Scan / import reporters. (TS mimic of brain_scan/reporters.py)
 *
 * Python `print` -> console.log. f-string field widths like `{tool:<12}` are
 * reproduced with String.padEnd. c.path.name -> path.basename(c.path).
 */
import * as path from "node:path";
import type { CandidateFile } from "./candidate.js";
import type { ImportResult } from "./importer.js";

/** Formats and prints the result of a Brain Scan. */
export function print_scan_report(candidates: CandidateFile[], as_json: boolean = false): void {
  if (as_json) {
    const data = candidates.map((c) => ({
      path: c.path,
      tool: c.tool,
      score: c.score_tier,
      size_mb: c.size_mb,
    }));
    console.log(JSON.stringify(data, null, 2));
    return;
  }

  console.log("\n🧠 NouGenShards Brain Scan\n");

  const high = candidates.filter((c) => c.score_tier === "high");
  const med = candidates.filter((c) => c.score_tier === "medium");

  const tools: Record<string, number> = {};
  for (const c of candidates) {
    tools[c.tool] = (tools[c.tool] ?? 0) + 1;
  }

  console.log("High-confidence AI memory:");
  for (const [tool, count] of Object.entries(tools)) {
    if (tool !== "unknown") {
      console.log(`  .${tool.padEnd(12)} found   ${count} files likely`);
    }
  }

  console.log("\nProject context:");
  for (const c of candidates.filter((c) => c.is_project_context).slice(0, 5)) {
    console.log(`  ${path.basename(c.path)}`);
  }
  if (candidates.filter((c) => c.is_project_context).length > 5) {
    console.log("  ... and more.");
  }

  console.log("\nSkipped danger zones:");
  console.log("  .ssh         skipped by default");
  console.log("  .aws         skipped by default");
  console.log("  .azure       skipped by default");
  console.log("  .config      skipped by default");

  console.log(`\nEstimated new shards: ${high.length * 2 + med.length}`); // Heuristic
  console.log("Cloud calls: 0");
  console.log("Files modified: 0");
  console.log("\nNothing imported.");
  console.log("Run: nougen brain import --confirm");
}

/** Formats and prints the result of a Brain Import. */
export function print_import_report(result: ImportResult, dry_run: boolean, as_json: boolean = false): void {
  if (as_json) {
    console.log(
      JSON.stringify(
        {
          dry_run: dry_run,
          files_scanned: result.files_scanned,
          records_parsed: result.records_parsed,
          shards_created: result.shards_created,
          duplicates_skipped: result.duplicates_skipped,
          secrets_redacted: result.secrets_redacted,
        },
        null,
        2,
      ),
    );
    return;
  }

  if (dry_run) {
    console.log("\n🧠 NouGenShards Brain Import (Dry Run)\n");
    console.log(`Files to scan: ${result.files_scanned}`);
    console.log(`Estimated records: ${result.records_parsed}`);
    console.log("\nRun: nougen brain import --confirm to write to memory.");
  } else {
    console.log("\n🧠 NouGenShards Brain Import Complete\n");
    console.log(`Files scanned:      ${result.files_scanned}`);
    console.log(`Records parsed:     ${result.records_parsed}`);
    console.log(`Shards created:     ${result.shards_created}`);
    console.log(`Duplicates skipped: ${result.duplicates_skipped}`);
    console.log(`Secrets redacted:   ${result.secrets_redacted}`);
    console.log("\n✅ Local memory enriched.");
  }
}
